presto-main/src/main/java/com/facebook/presto/sql/gen/BindCodeGenerator.java

```diff
@@ -39,8 +42,10 @@ public BindCodeGenerator(Map<LambdaDefinitionExpression, CompiledLambda> compile
     }
 
     @Override
-    public BytecodeNode generateExpression(Signature signature, BytecodeGeneratorContext context, Type returnType, List<RowExpression> arguments)
+    public BytecodeNode generateExpression(Signature signature, BytecodeGeneratorContext context, Type returnType, List<RowExpression> arguments, Optional<Variable> outputBlock)
     {
+        checkArgument(!outputBlock.isPresent());
```

Comment:
Add a comment that explains why this case can't happen

* `bind` is not top level in an expression.
* lambda cannot be put into blocks.